src/main.go

```diff
@@ -0,0 +1,103 @@
+package main
+
+import (
+    "log"
+    "net/http"
+	"fmt"
+	"os" 
+	"encoding/csv"
+	"encoding/json"
+	"strconv"
+)
+//Type to store records from CSV file
+type record struct {
+	Key int64
+	Value  string
+}
+
+var results []record
+
+// GetHandler handles the index route
+func GetGreetingHandler(w http.ResponseWriter, r *http.Request) {
+	w.Write([]byte("Welcome!, 2022Q2GO-Bootcamp"))
+}
+
+// GetCSVHandler handles the CSV route
+func GetCSVHandler(w http.ResponseWriter, r *http.Request) {
+	extractRecords("data.csv")
+	jsonBody, err := json.Marshal(results)
+	if err != nil {
+		http.Error(w, "Error converting results to json",
+			http.StatusInternalServerError)
+	}
+	w.Write(jsonBody)
+}
+
+
+// return data from file
+func extractRecords(fileName string) {
+	// open file
+    f, err := os.Open(fileName)
+    if err != nil {
+        log.Fatal(err)
+    }
+
+    // remember to close the file at the end of the program
+    defer f.Close()
+
+    // read csv values using csv.Reader
+    csvReader := csv.NewReader(f)
+    data, err := csvReader.ReadAll()
+    if err != nil {
+        log.Fatal(err)
```

Comment:
This error could be returned as a function result and managed as HTTP `Internal server Error` status to letting the user know that the data file was not found, this will keep the service running and, if the data file is fixed the service could recover and displays the data without restarting it